### PR TITLE
Clean fs

### PR DESCRIPTION
**Pull request by anton-bondarev**

### CODE REVIEW

File: src/compat/posix/fs/truncate.c [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * @author Denis Deryugin
+ */
+#include <unistd.h>
+#include <fcntl.h>
+#include <assert.h>
+
+int truncate(const char *path, off_t length) {
+	int fd, ret;
+
+	assert(path);
```
Comment (alexkalmuk) [inline src/compat/posix/fs/truncate.c]: It's not a good place for this `assert` in posix function.

Comment (anton-bondarev) [reply src/compat/posix/fs/truncate.c]: thanks, done

File: src/fs/driver/initfs/initfs_dvfs.c [hunk 2]
```diff
@@ -28,50 +29,19 @@
...
-/**
-* @brief Should be used as inode->i_data, but only for directories
-*/
-struct initfs_dir_info {
-	char  *path;
-	size_t path_len;
-	char  *name;
-	size_t name_len;
-};
+#define INITFS_MAX_NAMELEN 32
```
Comment (Deryugin) [inline src/fs/driver/initfs/initfs_dvfs.c]: This macro seems to be unused.

File: src/fs/driver/initfs/initfs_dvfs.c [hunk 3]
```diff
@@ -28,50 +29,19 @@
...
 
+POOL_DEF (initfs_file_pool, struct initfs_file_info, OPTION_GET(NUMBER,file_quantity));
```
Comment (Deryugin) [inline src/fs/driver/initfs/initfs_dvfs.c]: `POOL_DEF (`->`POOL_DEF(`

File: src/fs/driver/initfs/initfs_oldfs.c [hunk 4]
```diff
@@ -20,66 +20,31 @@
...
-	p_addr = data;
-
-	nas = desc->node->nas;
-	fi = (struct initfs_file_info *) nas->fi;
-
-	*p_addr = fi->addr;
-
-	return 0;
-}
+POOL_DEF (file_pool, struct initfs_file_info_tmp, OPTION_GET(NUMBER,file_quantity));
```
Comment (Deryugin) [inline src/fs/driver/initfs/initfs_oldfs.c]: `POOL_DEF (`->`POOL_DEF(`

File: src/fs/dvfs/dvfs.c [hunk 5]
```diff
@@ -453,7 +453,7 @@ int dvfs_mount(const char *dev, const char *dest, const char *fstype, int flags)
 
 		d->d_inode = dvfs_alloc_inode(sb);
 		*d->d_inode = (struct inode) {
-			.flags    = S_IFDIR,
+			.i_mode    = S_IFDIR,
```
Comment (Deryugin) [inline src/fs/dvfs/dvfs.c]: Please align `=` sign to other fields too

File: src/fs/dvfs/dvfs_util.c [hunk 6]
```diff
@@ -352,7 +352,7 @@ int dvfs_update_root(void) {
 
 	if (global_root->d_inode)
 		*(global_root->d_inode) = (struct inode) {
-			.flags    = S_IFDIR,
+			.i_mode    = S_IFDIR,
```
Comment (Deryugin) [inline src/fs/dvfs/dvfs_util.c]: Please align `=` sign to other fields

File: src/fs/inode.c [hunk 7]
```diff
@@ -1,3 +1,5 @@
+#include "inode.h"
```
Comment (Deryugin) [inline src/fs/inode.c]: This file is included at line 19

File: third-party/qt/tree/mkspecs/qws/embox-generic-g++/qplatformdefs.h [hunk 8]
```diff
@@ -71,10 +71,6 @@
 #define QT_NO_LOCALSERVER
```
Comment (Deryugin) [inline third-party/qt/tree/mkspecs/qws/embox-generic-g++/qplatformdefs.h]: Double blank line

File: third-party/qt/tree/mkspecs/qws/embox-generic-g++/qplatformdefs.h [hunk 9]
```diff
@@ -106,9 +102,6 @@
 #endif
```
Comment (Deryugin) [inline third-party/qt/tree/mkspecs/qws/embox-generic-g++/qplatformdefs.h]: Double blank line

File: third-party/qt/tree/mkspecs/qws/embox-generic-g++/qplatformdefs.h [hunk 10]
```diff
@@ -157,15 +126,11 @@ inline int symlink(const char *oldpath, const char *newpath) {
 }
```
Comment (Deryugin) [inline third-party/qt/tree/mkspecs/qws/embox-generic-g++/qplatformdefs.h]: Double blank line

File: src/fs/driver/initfs/initfs_fops.c [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+	off_t pos;
+
+	pos = file_get_pos(desc);
+	fi = file_get_inode_data(desc);
+
+	if (pos + size > file_get_size(desc)) {
+		size = file_get_size(desc) - pos;
+	}
+
+	memcpy(buf, (char *) (uintptr_t) (fi->start_pos + pos), size);
```
Comment (Deryugin) [inline src/fs/driver/initfs/initfs_fops.c]: Is this double cast `(char *) (uintptr_t)` really necessary?

Comment (anton-bondarev) [reply src/fs/driver/initfs/initfs_fops.c]: Yes, casting int to a pointer type isn't specified in standard

File: src/compat/posix/fs/chdir.c [hunk 12]
```diff
@@ -42,14 +36,12 @@ int chdir(const char *path) {
 		return -1;
 	}
 
-	/*check if such path exists in fs*/
-	if(0 != fs_perm_lookup(path, NULL, &last)){
-		SET_ERRNO(ENOENT);
+	if (stat(path, &buff)) {
+		/* SET_ERRNO(ENOENT); already set by stat()*/
```
[the remark this answers is not in the document]
Comment (anton-bondarev) [reply src/compat/posix/fs/chdir.c]: done

File: src/compat/posix/fs/oldfs/fsop.c [hunk 13]
```diff
@@ -10,14 +10,15 @@
...
 #include <fs/kfsop.h>
-#include <fs/node.h>
 #include <fs/vfs.h>
 #include <fs/perm.h>
 #include <fs/kfile.h>
 #include <sys/types.h>
 #include <stdio.h>
 
-#include "getumask.h"
+#include "fs/inode.h"
```
[the remark this answers is not in the document]
Comment (anton-bondarev) [reply src/compat/posix/fs/oldfs/fsop.c]: done

File: src/compat/posix/fs/open/open_dvfs.c [hunk 14]
```diff
@@ -43,7 +43,7 @@ int open(const char *path, int __oflag, ...) {
 	if (i_no == NULL)
 		return SET_ERRNO(ENOENT);
 
-	if ((i_no->flags & S_IFMT) == S_IFDIR) {
+	if ((i_no->i_mode & S_IFMT) == S_IFDIR) {
```
[the remark this answers is not in the document]
Comment (anton-bondarev) [reply src/compat/posix/fs/open/open_dvfs.c]: thanks. done

File: src/fs/driver/initfs/initfs_dvfs.c (hunk 2 above)
Comment (anton-bondarev) [reply src/fs/driver/initfs/initfs_dvfs.c]: done

File: src/fs/inode.c (hunk 7 above)
Comment (anton-bondarev) [reply src/fs/inode.c]: thanks. done

File: third-party/qt/tree/mkspecs/qws/embox-generic-g++/qplatformdefs.h (hunk 8 above)
Comment (anton-bondarev) [reply third-party/qt/tree/mkspecs/qws/embox-generic-g++/qplatformdefs.h]: done

File: third-party/qt/tree/mkspecs/qws/embox-generic-g++/qplatformdefs.h (hunk 9 above)
Comment (anton-bondarev) [reply third-party/qt/tree/mkspecs/qws/embox-generic-g++/qplatformdefs.h]: done

File: third-party/qt/tree/mkspecs/qws/embox-generic-g++/qplatformdefs.h (hunk 10 above)
Comment (anton-bondarev) [reply third-party/qt/tree/mkspecs/qws/embox-generic-g++/qplatformdefs.h]: done

File: src/fs/driver/devfs/devfs_dvfs.c [hunk 15]
```diff
@@ -50,7 +50,7 @@ void devfs_fill_inode(struct inode *inode,
 	assert(devmod);
 
 	inode->i_data = devmod;
-	inode->flags |= flags;
+	inode->i_mode |= (flags & S_IFMT);
```
[the remark this answers is not in the document]
Comment (anton-bondarev) [reply src/fs/driver/devfs/devfs_dvfs.c]: done

**Pull request merged**